src/test/java/org/assertj/core/api/Assertions_assertThat_with_Throwable_Test.java

```diff
@@ -52,11 +53,39 @@ public void should_fail_if_no_throwable_was_thrown() {
   @Test
   public void can_capture_exception_and_then_assert_following_AAA_or_BDD_style() {
     // when
-    Throwable boom = catchThrowable(raisingException("boom!!!!"));
+    Exception e = new Exception("boom!!");
+    Throwable boom = catchThrowable(raisingException(e));
 
     // then
-    assertThat(boom).isInstanceOf(Exception.class)
-                    .hasMessageContaining("boom");
+    assertThat(boom).isSameAs(e);
+  }
+
```

Comment:
add test when the code does not throw the expected throwable.

Reply:
This was a pre-existing test for `catchThrowable()`, which doesn't expect any particular throwable.

I modified it to make it stricter. The previous test could have passed if `catchThrowable()` returned a similar exception class to the one thrown; this one asserts that it must actually be the same exception.

Happy to add a test for `catchThrowable()` when the code doesn't throw an exception, but this is not strictly within the scope of my PR (though neither was this change, so I guess I brought it on myself!)

Reply:
Done.